### PR TITLE
Fix to return single latest record for Bar and Instrument from Parquet Catalog

### PR DESCRIPTION
**Pull request by rsmb7zhx**
# Pull Request

This is the fix I have made for duplicate records received for dataframe/nautilus objects and part is reported in #833 . This makes sure that the returned records are the latest single record being returned. Also subset keyword is added to allow custom unique filtering for any other classes doesn't added above. Feel free to adjust the code for any performance improvements.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How has this change been tested?

Catalog returns the unique Instruments after this change which otherwise shows multiple as described in #833.

### CODE REVIEW

File: nautilus_trader/persistence/catalog/parquet.py [hunk 1]
```diff
@@ -134,6 +148,7 @@ def _query(  # noqa (too complex)
                 table=table, mappings=mappings, raise_on_empty=raise_on_empty, **kwargs
             )
         else:
+            table = pa.Table.from_pandas(table)
```
Comment (rsmb7zhx) [inline nautilus_trader/persistence/catalog/parquet.py:151]: Although _handle_table_nautilus should accept the dataframe, however when passing dataframe directly following error occurs. If that is fixed then there is no need for this line.
```shell
currency_pair.pyx", line 201, in nautilus_trader.model.instruments.currency_pair.CurrencyPair.from_dict_c
            TypeError: Expected unicode, got float
```

**Comment by cjdsellers** [comment]
Thanks so much for the contribution!

I just had to change the base branch to `develop`, it looks like you may need to update the diff in some way to kickoff the `build_pr` workflow.

Also to save time, please ensure you've run the following over your branch:
```
pre-commit install
pre-commit run --all-files
```

**Comment by cjdsellers** [comment]
For some reason the `build_pr` workflow didn't run. I think because you initially diffed against the `master` branch.

This has happened before, and I expected changing the base branch would kick off the workflow.

I'll review this carefully tomorrow and merge anyway, and fix any issues which arise.

**Comment by cjdsellers** [comment]
So just a couple of things to fix, not too bad. Thanks again!

As above it would definitely have helped if you ran pre-commit first, and probably the tests locally too.
```
pre-commit install
pre-commit run --all-files
```

There's also a slightly move convenient make target:
```
make pre-commit
```

```
pytest tests
```

```
pre-commit run --all-files
forbid new submodules................................(no files to check)Skipped
fix python encoding pragma...............................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing nautilus_trader/persistence/catalog/parquet.py

debug statements (python)................................................Passed
detect private key.......................................................Passed
check builtin type constructor use.......................................Passed
check for case conflicts.................................................Passed
check that executables have shebangs.................(no files to check)Skipped
check json...............................................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check toml...............................................................Passed
check xml............................................(no files to check)Skipped
check yaml...............................................................Passed
codespell................................................................Passed
pycln (Python unused imports)............................................Passed
isort....................................................................Passed
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted nautilus_trader/persistence/catalog/parquet.py

All done! ✨ 🍰 ✨
1 file reformatted, 568 files left unchanged.

mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

nautilus_trader/persistence/catalog/parquet.py:127: error: Name "Instrument" is not defined
nautilus_trader/persistence/catalog/parquet.py:131: error: Name "Bar" is not defined
nautilus_trader/persistence/catalog/parquet.py:167: error: Cannot resolve name "df" (possible cyclic definition)
nautilus_trader/persistence/catalog/parquet.py:169: error: Cannot resolve name "df" (possible cyclic definition)
nautilus_trader/persistence/catalog/parquet.py:172: error: Cannot resolve name "df" (possible cyclic definition)
Found 5 errors in 1 file (checked 570 source files)

flake8 (Python)..........................................................Failed
- hook id: flake8
- exit code: 1

nautilus_trader/persistence/catalog/parquet.py:127:28: F821 undefined name 'Instrument'
nautilus_trader/persistence/catalog/parquet.py:131:21: F821 undefined name 'Bar'
nautilus_trader/persistence/catalog/parquet.py:167:13: F821 undefined name 'df'
nautilus_trader/persistence/catalog/parquet.py:167:30: F821 undefined name 'df'
nautilus_trader/persistence/catalog/parquet.py:169:12: F821 undefined name 'df'
nautilus_trader/persistence/catalog/parquet.py:172:18: F821 undefined name 'df'
6     F821 undefined name 'Instrument'

flake8 (Cython)..........................................................Passed
pydocstyle...............................................................Passed
nbstripout...............................................................Passed
Makefile:28: recipe for target 'pre-commit' failed
make: *** [pre-commit] Error 1

```

**Pull request merged**